astropy/__init__.py

```diff
@@ -6,6 +6,10 @@
 managing them.
 """
 
+# setuptools needs to be imported before distutils import happens
+# in various setup_package.py to avoid pytest collection error.
+import setuptools  # noqa
```

Comment:
Hmm... this makes setuptools a runtime dependency, does it? 💭

Reply:
Is this acceptable or do I need to hunt down all the distutils import and then import setuptools before each of them?